app/queries/servers/post.ts

```diff
@@ -16,7 +16,7 @@ export const prepareDeletePost = async (post: PostModel): Promise<Model[]> => {
     const relations: Array<Relation<Model> | Query<Model>> = [post.drafts, post.postsInThread];
     for await (const relation of relations) {
         try {
-            const model = await relation.fetch();
+            const model = await relation?.fetch();
```

Comment:
If the `relations` array is empty, it won't execute the for loop.

Reply:
Why would `relations` be empty? it contains `drafts` and `postsInThread` see the `relations` variable declaration
`const relations: Array<Relation<Model> | Query<Model>> = [post.drafts, post.postsInThread];`